plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -113,12 +113,32 @@ protected final void commitNewTable(TableMetadata metadata)
         try {
             metastore.createTable(table, privileges);
         }
-        catch (SchemaNotFoundException
-               | TableAlreadyExistsException e) {
+        catch (SchemaNotFoundException e) {
             // clean up metadata files corresponding to the current transaction
             fileIo.deleteFile(newMetadataLocation);
             throw e;
         }
+        catch (TableAlreadyExistsException e) {
```

Comment:
potential follow-up: Please add a similar handling in `io.trino.plugin.iceberg.catalog.glue.GlueIcebergTableOperations#commitNewTable`